### ISSUE TITLE
rest.index/refresh fails for Elasticsearch 5

### ISSUE THREAD

**Issue by hantuzun** [clojurewerkz/elastisch#257]
When I updated Elasticsearch to 5.5.2, our `rest.index/refresh` calls started to fail with a message from `clj-http` that "text/plain" content type is not acceptable.

Updating `clojurewerkz.elastisch.rest`'s `post` function with a `:content-type` `:json`  setting fixed the issue for me, without breaking other hundreds of tests using elastisch for our application:

```clj
(defn post
  ([^Connection conn ^String uri]
   (post conn uri {}))
  ([^Connection conn ^String uri {:keys [body] :as options}]
    (-> uri
        (http/post (merge (.http-opts conn)
                          options
                          {:accept :json
                           :content-type :json
                           ;:throw-exceptions false ;;ables to see ES when debugging
                           :body (json/encode body)}))
        (:body)
        (parse-safely))))
```

**Comment by michaelklishin** [comment]
See [Supported ElasticSearch versions](https://github.com/clojurewerkz/elastisch/blob/master/README.md#supported-elasticsearch-versions) and #245.

**Comment by michaelklishin** [comment]
@hantuzun if you'd like to contribute this small improvement, a pull request against master would be appreciated.

**Comment by hantuzun** [comment]
Hi @michaelklishin, I fear that change might break some features in some of the many Elasticsearch versions supported.

**Comment by michaelklishin** [comment]
@hantuzun I don't see how adding a content type can break anything. In any case, no PRs for 5.x will be merged until 3.0 is released.

**Comment by hantuzun** [comment]
I've already forked elastisch to my account and we're using it. It'd be a hassle to fork it again for this PR that's not merged.

Hopefully 3.0 could be released soon and we could work on Elasticsearch 5.x support.
